ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/EthProtocol.java

```diff
@@ -46,8 +46,17 @@ public static boolean requestIdCompatible(final int code) {
             EthProtocolMessages.GET_NODE_DATA,
             EthProtocolMessages.NODE_DATA,
             EthProtocolMessages.GET_RECEIPTS,
-            EthProtocolMessages.RECEIPTS)
-        .contains(code);
+            EthProtocolMessages.RECEIPTS);
+    REQUEST_ID_MESSAGES =
+        new BitSet(requestIdMessages.stream().mapToInt(i -> i).max().getAsInt() + 1);
```

Comment:
is this more optimal than just getting the size of the set?

Reply:
We can make it even constant, like 64 which is the size of a long (64 bits).

Reply:
the requestIdMessage is the index of the bitset, so the bitset should be large enough for the message with the largest index, since bitset starts from 0 then the optimal size is max index + 1, not sure what do you mean by decoupling?
Then I plan to make `EthProtocolMessages` an enum, so we can directly use EnumSet that is clearer.